.github/workflows/macos-build.yml

```diff
@@ -0,0 +1,574 @@
+    - name: Build Mksnapshot (darwin)
+      run: |
+        cd src
+        e build electron:electron_mksnapshot -j $NUMBER_OF_NINJA_PROCESSES
+        gn desc out/Default v8:run_mksnapshot_default args > out/Default/mksnapshot_args
+        # Remove unused args from mksnapshot_args
+        SEDOPTION="-i ''"
+        sed $SEDOPTION '/.*builtins-pgo/d' out/Default/mksnapshot_args
+        sed $SEDOPTION '/--turbo-profiling-input/d' out/Default/mksnapshot_args
+        sed $SEDOPTION '/The gn arg use_goma=true .*/d' out/Default/mksnapshot_args
+        if [ "$SKIP_DIST_ZIP" != "1" ]; then
+          e build electron:electron_mksnapshot_zip -j $NUMBER_OF_NINJA_PROCESSES
+          (cd out/Default; zip mksnapshot.zip mksnapshot_args gen/v8/embedded.S)
+        fi
+    - name: Build Chromedriver (darwin)
+      run: |
+        cd src
+        e build electron:electron_chromedriver -j $NUMBER_OF_NINJA_PROCESSES
+        e build electron:electron_chromedriver_zip
+    - name: Build Node Headers (darwin)
+      run: |
+        cd src
+        e build electron:node_headers
+    - name: Generate & Zip Symbols (darwin)
+      run: |
+        if [ "$GENERATE_SYMBOLS" == "true" ]; then
+          e build electron:electron_symbols
+        fi
+        cd src
+        export BUILD_PATH="$(pwd)/out/Default"
+        e build electron:licenses
+        e build electron:electron_version_file
+        electron/script/zip-symbols.py -b $BUILD_PATH
+      # TODO(vertedinde): handle creating ffmpeg and hunspell for release builds
+      # The current generated_artifacts_<< artifact.key >> name was taken from CircleCI
+      # tp ensure we don't break anything, but we may be able to improve that.
+    - name: Move all Generated Artifacts to Upload Folder
+      run: ./src/electron/script/actions/move-artifacts.sh
+    - name: Upload Generated Artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: generated_artifacts_darwin
+        path: ./generated_artifacts_darwin
+    - name: Persist Build Artifacts
+      uses: actions/cache/save@v4
+      with:
+        path: |
+          src/out/Default/gen/node_headers
+          src/out/Default/overlapped-checker
+          src/electron
+          src/third_party/electron_node
+          src/third_party/nan
+          src/cross-arch-snapshots
+          src/third_party/llvm-build
+          src/build/linux
+          src/buildtools/mac
+          src/buildtools/third_party/libc++
+          src/buildtools/third_party/libc++abi
+          src/third_party/libc++
+          src/third_party/libc++abi
+          src/out/Default/obj/buildtools/third_party
+          src/v8/tools/builtins-pgo
+        key: ${{ runner.os }}-build-artifacts-darwin-${{ github.sha }}
+    - name: Create MAS Config
+      run: |
+        mv src/electron/script/actions/evm.mas.json $HOME/.electron_build_tools/configs/evm.mas.json
+        echo "MAS_BUILD=true" >> $GITHUB_ENV
+        e use mas
+    - name: Build Electron (mas)
```

Comment:
The fact that none of this can be re-used is legitimately upsetting to me

Reply:
I know, it's legitimately very frustrating - I'm looking into a reusable workflow for publishing that reuses the build workflow. If that works, we may be able to use similar logic for some of these build steps.